presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java

```diff
@@ -99,4 +113,30 @@ public void testNationJoinRegion()
 
         assertThat(queryResult).containsOnly(row("CANADA", "AMERICA"));
     }
+
+    @Test(groups = SQL_SERVER)
+    public void testAllDatatypes()
+            throws SQLException
+    {
+        String sql = format(
+                "SELECT bi, si, i, ti, f, r, c, vc, te, nc, nvc, nt, d, dt, dt2, sdt " +
+                        "FROM %s.%s.%s", CONNECTOR_NAME, KEY_SPACE, SQLSERVER_ALL_TYPES.getName());
+        QueryResult queryResult = onPresto()
+                .executeQuery(sql);
+
+        assertThat(queryResult)
+                .hasColumns(BIGINT, SMALLINT, INTEGER, TINYINT, DOUBLE, REAL, CHAR, LONGNVARCHAR, LONGNVARCHAR,
+                        CHAR, LONGNVARCHAR, LONGNVARCHAR, DATE, TIMESTAMP, TIMESTAMP, TIMESTAMP)
+                .containsOnly(
+                        row(Long.MIN_VALUE, Short.MIN_VALUE, Integer.MIN_VALUE, Byte.MIN_VALUE, Double.MIN_VALUE,
+                                Float.MIN_VALUE, "\0   ", "\0", "\0", "\0   ", "\0", "\0",
+                                Date.valueOf("0001-01-02"), Timestamp.valueOf("1753-01-01 00:00:00.000"),
+                                Timestamp.valueOf("0001-01-01 00:00:00.000"), Timestamp.valueOf("1900-01-01 00:00:00")),
```

Comment:
This is supported by respective SQL Server data type. For e.g. for date
`0001-01-01 through 9999-12-31 `
https://msdn.microsoft.com/en-us/library/bb630352.aspx